src/Components/Blazor/WebAssembly.Authentication/src/AuthenticationManager.cs

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// A component that handles remote authentication operations in an application.
+    /// </summary>
+    /// <typeparam name="TAuthenticationState">The user state type persisted while the operation is in progress. It must be serializable.</typeparam>
+    public class AuthenticationManager<TAuthenticationState> : ComponentBase where TAuthenticationState : RemoteAuthenticationState
+    {
+        private string _message;
+
+        /// <summary>
+        /// Gets or sets the <see cref="RemoteAuthenticationActions"/> action the component needs to handle.
+        /// </summary>
+        [Parameter] public string Action { get; set; }
```

Comment:
I assume you looked into the possibility of making `RemoteAuthenticationActions` into an enum, did you? Is there some technical reason (such as lack of router support) why it needs to be a `string`?

Reply:
Currently being tracked by https://github.com/dotnet/aspnetcore/issues/19054